beamer/agent/chain.py

```diff
@@ -548,6 +549,8 @@ def get_l1_cost(context: Context) -> int:
 
 def _l1_resolution_threshold_reached(claim: Claim, context: Context) -> bool:
     limit = get_l1_cost(context)
+    if context.target_chain_id in [10, 420, 17] and not claim.message_proved:
```

Comment:
> It seems misplaced to me to do this check here. This function is purely about a financial limit. I would rather do this check somewhere else.

Definitely. I would move this check to `maybe_prove`, with this being literally the first line in it.
So that `maybe_prove` does not do anything if it's not Optimisim or the "message" is already proved.